src/main.cpp

```diff
@@ -1083,6 +1084,12 @@ bool CheckTransactionWithoutProofVerification(const CTransaction& tx, CValidatio
                              REJECT_INVALID, "bad-txns-txouttotal-toolarge");
     }
 
+    // Check for overflow valueBalance
+    if (std::abs(tx.valueBalance) > MAX_MONEY) {
```

Comment:
Let's not use `std::abs`...

> The behavior is undefined if the result cannot be represented by the return type. 

Let's just mask off the sign bit and check if it's `> MAX_MONEY`?